pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1490,6 +1490,26 @@ func (c *Cloud) getAwsInstance(nodeName types.NodeName) (*awsInstance, error) {
 	return awsInstance, nil
 }
 
+// wrapAttachError wraps the error returned by an AttachVolume request with
+// additional information, if needed and possible.
+func wrapAttachError(err error, disk *awsDisk, instance string) error {
+	if awsError, ok := err.(awserr.Error); ok {
+		if awsError.Code() == "VolumeInUse" {
+			info, err := disk.describeVolume()
+			if err == nil {
+				for _, a := range info.Attachments {
+					if a.VolumeId == nil || a.State == nil || a.InstanceId == nil {
+						continue
+					} else if *disk.awsID.awsString() == *a.VolumeId && *a.State == "attached" {
```

Comment:
I think also we want to glog if we get back a different volumeId (given we are querying by volume id), and we probably want to log regardless of attachment state?